types/node/v16/ts4.8/fs.d.ts

```diff
@@ -3392,19 +3393,31 @@ declare module 'fs' {
     export function accessSync(path: PathLike, mode?: number): void;
     interface StreamOptions {
         flags?: string | undefined;
-        encoding?: BufferEncoding | undefined;
-        fd?: number | promises.FileHandle | undefined;
+        encoding?: BufferEncoding | null | undefined;
+        fd?: number | promises.FileHandle | null | undefined;
         mode?: number | undefined;
         autoClose?: boolean | undefined;
-        /**
-         * @default false
-         */
         emitClose?: boolean | undefined;
         start?: number | undefined;
-        highWaterMark?: number | undefined;
+    }
+    interface FSImplementation {
+        open?: Function;
```

Comment:
As far as I can tell, `Function` is not used in other places; this type is generally to be avoided in favor of something like `(...args: any[]) => any` or similar because `Function` can accidentally be `new`'d among other problems.

Is there no concrete type for this?

Was the intent of https://github.com/DefinitelyTyped/DefinitelyTyped/pull/66054#issuecomment-1633913429 because these functions have properties you don't want?

Reply:
Ideally `open` in `FSImplementation` should have the same type as `open` in `node:fs` (and maybe `open` in `node:fs/promises` also, need to check). But `open` is overloaded function and I hope that we wan't to copy-paste all function declarations and support it later.

> Was the intent of https://github.com/DefinitelyTyped/DefinitelyTyped/pull/66054#issuecomment-1633913429 because these functions have properties you don't want?

Yes, we have also namespace `open` with `__promisify__`, that is why we can't just use `typeof open`.

Reply:
Well, you could `Omit<typeof open, "__promisify__">` if there are no other props. `Function` just doesn't seem like the best choice overall.

Reply:
No, `Omit` doesn't work here, because `open` is namespace. `Omit<typeof open, "__promisify__">` is the same as `{}`.

Reply:
Dang, you're right; I didn't test the negative case.

Reply:
Given that, then I guess you can just do an any function, but I don't think `Function` is the right thing to use for this.

Reply:
Yes, if you don't want `new`-able things to be assigned to it: [Playground Link](https://www.typescriptlang.org/play?#code/CYUwxgNghgTiAEEQBd5igBwJbKhAYgK4B2YyWA9sQFzxGnlUDcAUKJLAkqrDBQO70ylGvAAUAOimwA5gGdaUYgE8A2gF0AlPAC8APnhLlrFiw5y58AILwA3gF824aHHgAzEsKrvPYgB60xIQAtgBGIDAANPDKgSHhMJpxYREmpujYuASejMS61qwZOHhCufkepCa8AqUi+Vas1YI5dTo+lUA)